pkg/lspext/lspext.go

```diff
@@ -1,47 +1,80 @@
 package lspext
 
-import "github.com/sourcegraph/go-langserver/pkg/lsp"
+import (
+	"fmt"
+	"sort"
+	"strings"
 
-// WorkspaceReferenceParams is a parameter literal used in `workspace/reference`
-// requests. This is a Sourcegraph extension method to LSP. It is sent from the
-// client to the server, and the response type is `[]ReferenceInformation`.
-//
-// It strictly returns the locations in the workspace at which symbols outside
-// of the workspace are referenced. That is:
+	"github.com/sourcegraph/go-langserver/pkg/lsp"
+)
+
+// WorkspaceReferencesParams is parameters for the `workspace/xreferences` extension
 //
-// 	- Excluding any `URI` which is located within the workspace.
-// 	- Excluding any `Name` which is unexported or private (i.e. according to
-// 	  language semantics).
-// 	- Excluding any `Location` which is located in vendored code (e.g.
-// 	  `vendor/...` for Go, `node_modules/...` for JS, .tgz NPM packages, or
-// 	  .jar files for Java).
+// See: https://github.com/sourcegraph/language-server-protocol/blob/master/extension-workspace-reference.md
 //
-type WorkspaceReferenceParams struct {
-	Limit int `json:"limit"`
+type WorkspaceReferencesParams struct {
 }
 
+// ReferenceInformation is the array response type for the `workspace/xreferences` extension
+//
+// See: https://github.com/sourcegraph/language-server-protocol/blob/master/extension-workspace-reference.md
+//
 type ReferenceInformation struct {
-	// Location is the location at which Symbol has been referenced.
-	Location lsp.Location `json:"location"`
-
-	// Name is the name of the symbol that is being referenced. For example:
-	//
-	// 	(Go) "ServeHTTP"
-	// 	(JS) "render"
-	//
-	Name string `json:"name"`
-
-	// ContainerName is the container name of the symbol that is being
-	// referenced. For example:
-	//
-	// 	(Go) "Router"
-	// 	(JS) "ReactMount"
-	//
-	ContainerName string `json:"containerName,omitempty"`
-
-	// URI is the URI location of the symbol that is being referenced. If both
-	// Name and ContainerName are empty strings, it implies that the URI
-	// location is referenced but not a particular symbol (e.g. a Go import
-	// statement, a JS require() call, etc).
-	URI string `json:"uri"`
+	Reference lsp.Location     `json:"reference"`
+	Symbol    SymbolDescriptor `json:"symbol"`
+}
+
+type SymbolDescriptor struct {
+	Name          string                 `json:"name,omitempty"`
+	Kind          lsp.SymbolKind         `json:"kind,omitempty"`
+	File          string                 `json:"file,omitempty"`
+	ContainerName string                 `json:"containerName,omitempty"`
+	Vendor        bool                   `json:"vendor,omitempty"`
+	Meta          map[string]interface{} `json:"meta,omitempty"`
```

Comment:
If `SymbolDescriptor` itself is "metadata", having a `Meta` field inside it seems redundant. What about something like `Attributes`?